### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by fulf**

### CODE REVIEW

File: lib/faraday_oauth2_ccg_middleware.rb [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+  # into the Authorization header
+  #
+  # @example Configure OAUTH Client Credentials Grant middleware
+  # Faraday.new do |conn|
+  #   conn.request :oauth2_ccg,
+  #                oauth_host:    'https://server.example.com',
+  #                token_url:     '/token',
+  #                client_id:     's6BhdRkqt3',
+  #                client_secret: '7Fjfp0ZBr1KtDRbnfVdmIw',
+  #                cache_store:   ::ActiveSupport::Cache.lookup_store(:redis_store, 'redis://127.0.0.1:6379')
```
Comment (fulf) [inline lib/faraday_oauth2_ccg_middleware.rb:21]: I'm not sure on how to handle the caching. I see two ways:

1. Add `ActiveSupport` gem dependency and allow the user to pass just the configuration.
- enforces cache store interface
- extra dependency

2. Allow the user to pass an already configured cache store instance
- no extra dependency
- cannot enforce its interface meaning `.fetch(key, *options, &blk)` may fail

Comment (ovidiubite) [reply lib/faraday_oauth2_ccg_middleware.rb:21]: I would say that the second option is a better one. 
Maybe you could just throw an CacheConfigurationError if the cache store has a different interface that the one we are expecting. 

You could also whitelist the stores that are supported?
Another option is to add a new option that says what store it is (not only the instance) and based on that to have multiple implementation of the caching mechanism. But I would not go too far with this, unless it's needed

Comment (ovidiubite) [reply lib/faraday_oauth2_ccg_middleware.rb:21]: like:
```
cache_store_instance:   ::ActiveSupport::Cache.lookup_store(:redis_store, 'redis://127.0.0.1:6379')
cache_store_type: :active_support
```

Comment (International) [reply lib/faraday_oauth2_ccg_middleware.rb:21]: for example activestorage also does not depend explicitly on everything: https://github.com/rails/rails/blob/master/activestorage/activestorage.gemspec , in their service impl they add the dep there: https://github.com/rails/rails/blob/master/activestorage/lib/active_storage/service/s3_service.rb

**Pull request merged**